### PR TITLE
[doc] Enhance documentation about performance log directories

### LINKED ISSUE #1643: Enhance documentation about performance log directories
According to feedback from the Slack channel it can be a bit confusing. We must explain it better and give some concrete examples.

### PR DESCRIPTION
**Pull request by rsarm**
Fixes #1643.

### CODE REVIEW

File: docs/configure.rst [hunk 1]
```diff
@@ -192,6 +192,9 @@ For example, if we have set it to ``warning``, no debug or informational message
 Finally, there is a special set of handlers for handling performance log messages.
 These are stored in the ``handlers_perflog`` property.
 The performance handler in this example will create a file per test and per system/partition combination and will append the performance data to it every time the test is run.
+The path to the file is created as follows: ``<basedir>/<prefix>/<testname>log``.
+``<basedir>`` is controlled by the ``--perflogdir`` option, the environment variable :envvar:`RFM_PERFLOG_DIR` or the ``basedir`` configuration parameter in the ``handlers_perflog``, which is ``.`` by default; ``<prefix>`` is what is defined in the ``handlers_perflog``.
+Note that that the performance log file is created only if ``perf_patterns`` is defined on the test.
```
Comment (teojgo) [inline docs/configure.rst:197]:
```suggestion
Note that the performance log file is created only if ``perf_patterns`` is defined on the test.
```

File: docs/configure.rst [hunk 2]
```diff
@@ -192,6 +192,9 @@ For example, if we have set it to ``warning``, no debug or informational message
 Finally, there is a special set of handlers for handling performance log messages.
 These are stored in the ``handlers_perflog`` property.
 The performance handler in this example will create a file per test and per system/partition combination and will append the performance data to it every time the test is run.
+The path to the file is created as follows: ``<basedir>/<prefix>/<testname>log``.
+``<basedir>`` is controlled by the ``--perflogdir`` option, the environment variable :envvar:`RFM_PERFLOG_DIR` or the ``basedir`` configuration parameter in the ``handlers_perflog``, which is ``.`` by default; ``<prefix>`` is what is defined in the ``handlers_perflog``.
```
Comment (teojgo) [inline docs/configure.rst:196]:
```suggestion
``<basedir>`` is controlled by the ``--perflogdir`` option, the environment variable :envvar:`RFM_PERFLOG_DIR` and the ``basedir`` configuration parameter in the ``handlers_perflog``, which is ``.`` by default; ``<prefix>`` is defined in the ``handlers_perflog``.
```

**Review by teojgo — CHANGES_REQUESTED** [review]

File: docs/configure.rst [hunk 3]
```diff
@@ -192,6 +192,9 @@ For example, if we have set it to ``warning``, no debug or informational message
 Finally, there is a special set of handlers for handling performance log messages.
 These are stored in the ``handlers_perflog`` property.
 The performance handler in this example will create a file per test and per system/partition combination and will append the performance data to it every time the test is run.
```
Comment (vkarak) [inline docs/configure.rst:194]:
```suggestion
The performance handler in this example will create a file per test and per system/partition combination and will append the performance data to it every time the *performance test* is run.
```

Then either we define what a performance test is or have link to somewhere that we define it.

File: docs/configure.rst (hunk 2 above)
Comment (vkarak) [reply docs/configure.rst:196]: Imho, all this description is not needed. It described in the configuration reference of the `filelog` handler. So this is a duplication. We do have a link to the reference at the end of this section. The key problem in the documentation, in my opinion, is that we don't say explicitly that performance logging happens *only* for performance tests.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Review by vkarak — APPROVED** [review]

**Pull request merged**